eng/scripts/Language-Settings.ps1

```diff
@@ -5,36 +5,82 @@ $packagePattern = "*.pom"
 $MetadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/java-packages.csv"
 $BlobStorageUrl = "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=java%2F&delimiter=%2F"
 
-function Get-java-PackageInfoFromRepo ($pkgPath, $serviceDirectory, $pkgName)
+function Get-java-PackageInfoFromRepo
 {
-  $projectPath = Join-Path $pkgPath "pom.xml"
-
-  if (Test-Path $projectPath)
+  [CmdletBinding()]
+  Param(
+    [Parameter(Position = 0)]
+    [string]$pkgDirectoryPath,
+    [Parameter(Position = 1)]
+    [string]$serviceDirectoryName,
+    $pkgPath,
+    $serviceDirectory,
+    $pkgName
+  )
+
+  if ($pkgName)
   {
-    $projectData = New-Object -TypeName XML
-    $projectData.load($projectPath)
-    $projectPkgName = $projectData.project.artifactId
-    $pkgVersion = $projectData.project.version
-    $pkgGroup = $projectData.project.groupId
+    $projectPath = Join-Path $pkgPath "pom.xml"
+    if (Test-Path $projectPath)
+    {
+      $projectData = New-Object -TypeName XML
+      $projectData.load($projectPath)
+      $projectPkgName = $projectData.project.artifactId
+      $pkgVersion = $projectData.project.version
+      $pkgGroup = $projectData.project.groupId
 
-    if ($projectPkgName -eq $pkgName)
+      if ($projectPkgName -eq $pkgName)
```

Comment:
Shouldn't this be the only check we need to remove if pkgName is null? We shouldn't need to duplicate the entire block.

Reply:
Yes but if you look am also making using the opportunity to make the argument names clearer. So `pkgDirectoryPath` rather than `pkgPath`